spec/job_spec.rb

```diff
@@ -16,4 +18,37 @@
     expect(obj.name).to    be_a String
     expect(obj.related).to be_a AnsibleTowerClient::Job::Related
   end
+
+  describe "#extra_vars_hash" do
+    describe "#extra_vars exists" do
+      it "returns a hashed value" do
+        obj = described_class.new(instance_double("AnsibleTowerClient::Api"), raw_instance)
+        expect(obj.extra_vars_hash).to eq('option' => 'lots of options')
+      end
+    end
+
+    describe "#extra_vars does not exist" do
+      it "returns an empty hash" do
+        obj = described_class.new(instance_double("AnsibleTowerClient::Api"), raw_instance_no_extra_vars)
+        expect(obj.extra_vars_hash).to be_a_kind_of(Hash)
```

Comment:
You could `expect(obj.extra_vars_hash).to eq({})` to reduce the number of expectations.